library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -76,6 +76,35 @@ class Client() {
         fun register(codec: ContentCodec<*>) {
             codecRegistry.register(codec = codec)
         }
+
+        suspend fun canMessage(accountAddresses: List<String>, appContext: Context, api: ClientOptions.Api): Map<String, Boolean> {
+            val accountAddress = "0x0000000000000000000000000000000000000000"
+            val inboxId = getOrCreateInboxId(api, accountAddress)
+            val alias = "xmtp-${api.env}-$inboxId"
+
+            val directoryFile = File(appContext.filesDir.absolutePath, "xmtp_db")
+            directoryFile.mkdir()
+            val dbPath = directoryFile.absolutePath + "/$alias.db3"
+
+            val ffiClient = createClient(
+                logger = XMTPLogger(),
+                host = api.env.getUrl(),
+                isSecure = api.isSecure,
+                db = dbPath,
+                encryptionKey = null,
+                accountAddress = accountAddress.lowercase(),
+                inboxId = inboxId,
+                nonce = 0.toULong(),
+                legacySignedPrivateKeyProto = null,
+                historySyncUrl = null
+            )
+
+            val result = ffiClient.canMessage(accountAddresses)
+            ffiClient.releaseDbConnection()
+            File(dbPath).delete()
```

Comment:
@rygine I think we will want to delete the db that is created with this temp client? Maybe something to add to the js sdk.

Reply:
yep! i recently added the ability to create a browser/node clients without a DB or encryption key for this purpose. just need to update the SDKs. great reminder, thanks!